docs/contributing/database-migrations/edd.mdx

```diff
@@ -173,49 +222,82 @@ END
 </TabItem>
 </Tabs>
 
-## Workflow
+## Our EDD Process
+
+There are some unique constraints to Bitwarden that are not addressed directly in Martin Fowler's
+EDD article.
+
+- Our Production instances in the cloud are required to be on at all times
+- Our self-host instances that we do not directly have access to manage must support the same EDD
+  processes; however, they do not have the same always-on application constraint
+- Minimization of manual steps in our process
+
+The process to support all of these constraints is a complex one. Below is an image of a state
+machine that will hopefully help visualize the process and what it supports. It assumes that all
+database changes follow the standards that are laid out in [Migrations](./).
 
-The Bitwarden specific workflow for writing migrations are described below.
+---
 
-### Developer
+![Bitwarden EDD State Machine](./edd_state_machine.jpg) \[Open Image in a new tab for better
+viewing\]
 
-The development flow is described in [Migrations](./).
+---
 
-### Devops
+### Cloud Environments
```

Comment:
💭 Is "cloud" the term we want? Or perhaps more generically "online"?

Reply:
I went with "always-on" over "online" to avoid the ambiguous "online application" possibly referring to any application on the internet. Thoughts?